debezium-core/src/main/java/io/debezium/transforms/time/TimestampConverter.java

```diff
@@ -0,0 +1,785 @@
+            if (!VALID_TYPES.contains(typeCheck)) {
+                throw new ConfigException("Unknown timestamp type in TimestampConverter: " + typeCheck + ". Valid values are "
+                        + Utils.join(VALID_TYPES, ", ") + ".");
+            }
+            if (typeCheck.equals(TYPE_STRING) && formatPatternTimestamp.trim().isEmpty()) {
+                throw new ConfigException("TimestampConverter requires format option to be specified when using string timestamps");
+            }
+            if (typeCheck.equals(TYPE_STRING) && formatPatternDate.trim().isEmpty()) {
+                throw new ConfigException("TimestampConverter requires format option to be specified when using string dates");
+            }
+        }
+
+        SimpleDateFormat format = null;
+        SimpleDateFormat dateFormat = null;
+        if (formatPatternTimestamp != null && !formatPatternTimestamp.trim().isEmpty()) {
+            try {
+                format = new SimpleDateFormat(formatPatternTimestamp);
+                format.setTimeZone(UTC);
+            } catch (IllegalArgumentException e) {
+                throw new ConfigException("TimestampConverter requires a SimpleDateFormat-compatible pattern for string timestamps: "
+                        + formatPatternTimestamp, e);
+            }
+        }
+        if (formatPatternDate != null && !formatPatternDate.trim().isEmpty()) {
+            try {
+                dateFormat = new SimpleDateFormat(formatPatternDate);
+                dateFormat.setTimeZone(UTC);
+            } catch (IllegalArgumentException e) {
+                throw new ConfigException("TimestampConverter requires a SimpleDateFormat-compatible pattern for string dates: "
+                        + formatPatternDate, e);
+            }
+        }
+        config = new Config(field, format, dateFormat, field_type);
+    }
+
+    @Override
+    public R apply(R record) {
+        if (operatingSchema(record) == null) {
+            return applySchemaless(record);
+        } else {
+            return applyWithSchema(record);
+        }
+    }
+
+    @Override
+    public ConfigDef config() {
+        return FIELD_DOC;
+    }
+
+    @Override
+    public void close() {
+    }
+
+    public static class Key<R extends ConnectRecord<R>> extends TimestampConverter<R> {
+        @Override
+        protected Schema operatingSchema(R record) {
+            return record.keySchema();
+        }
+
+        @Override
+        protected Object operatingValue(R record) {
+            return record.key();
+        }
+
+        @Override
+        protected R newRecord(R record, Schema updatedSchema, Object updatedValue) {
+            return record.newRecord(record.topic(), record.kafkaPartition(), updatedSchema, updatedValue, record.valueSchema(), record.value(), record.timestamp());
+        }
+    }
+
+    public static class Value<R extends ConnectRecord<R>> extends TimestampConverter<R> {
```

Comment:
Let's extract `Key`  and `Value` into separate source files. That way people won't struggle with specifying an inner class name when configuring this SMT.

Reply:
Key and Value classes are static , i would like to know the way, you prefer to separate them.
If you can provide any example, that you think is optimal.

Reply:
@gunnarmorling ^^

Reply:
Technically, the static inner classes are fine. It's just a usability concern to remember whether the name must be given as `io.debezium.transforms.time.TimestampConverter.Value` or as `io.debezium.transforms.time.TimestampConverter$Value`. That's avoided by not requiring the user to specify any inner classes.